rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -219,11 +219,11 @@ private boolean validateLogsBloom(BlockHeader header, BlockResult result) {
 
     @VisibleForTesting
     public BlockResult execute(Block block, BlockHeader parent, boolean discardInvalidTxs) {
-        return execute(block, parent, discardInvalidTxs, false);
+        return execute(block, parent, discardInvalidTxs, false, true);
     }
 
-    public BlockResult execute(Block block, BlockHeader parent, boolean discardInvalidTxs, boolean ignoreReadyToExecute) {
-        return executeInternal(null, 0, block, parent, discardInvalidTxs, ignoreReadyToExecute);
+    public BlockResult execute(Block block, BlockHeader parent, boolean discardInvalidTxs, boolean ignoreReadyToExecute, boolean saveState) {
```

Comment:
This method is receiving a bunch of input parameters. For the future if this continue growing we could create something like a BlockExecutorParameters and collect them there to avoid the method signature continue growing, and avoiding possible errors in the order of the passed parameters. 

Adding the Builder pattern to the ParametersClass we could have something like:
```
BlockExecutorParameters.builder()
                .discardInvalidTx(true)
                .ignoreReadyToExecute(true)
                .ignoreSaveState(true)
                .build;
```
For me is easier to read when coding. Also we could create specific methods to create specific "configurations" for example a default() or other different ones and use then if the same behaviour is expected when calling executor from different places avoiding duplication.